pkg/loader/loader.go

```diff
@@ -329,7 +331,7 @@ func LoadRoots(roots ...string) ([]*Package, error) {
 //
 // This is generally only useful for use in testing when you need to modify
 // loading settings to load from a fake location.
-func LoadRootsWithConfig(cfg *packages.Config, roots ...string) ([]*Package, error) {
+func LoadRootsWithConfig(cfg *packages.Config, roots ...string) (pkgs []*Package, retErr error) {
```

Comment:
Just to summarize what we do now:
* If four dot notation, traverse and find go.mod files. If found, add to goModDirs
* Load as we already do today
* Iterate go.mod dirs, chdir into them, load ./...
* chdir back to not cause weirdness
Is that dorrect?

Reply:
You nailed it.